### ISSUE TITLE
Numerical instability in long tunnel simulation

### ISSUE THREAD

**Issue by mcgratta** [firemodels/fds#11428]
This issue started as a [Discussion Group post](https://groups.google.com/g/fds-smv/c/hwzAtduEtqg/m/fwngujV3GQAJ?utm_medium=email&utm_source=footer)

 I'm trying now to simulate a ;pmg tunnel, and it keeps crashing. I tried :

TUNNEL_PRECONDITIONNER=.T. 
MAX_PRESSURE_ITERATIONS=100
VELOCITY_TOLERANCE=0.001
Everytime it is the same : Warnings about minimum density, and a few steps later numerical instability occurs.

I know I could create some small holes, but I know that my simulation will take forever with it...

Any other suggestions?
[Tunnel.txt](https://github.com/firemodels/fds/files/10679506/Tunnel.txt)

**Comment by mcgratta** [comment]
[Tunnel2.txt](https://github.com/firemodels/fds/files/10686454/Tunnel2.txt)
[Tunnel4.txt](https://github.com/firemodels/fds/files/10686455/Tunnel4.txt)
I am currently running two variations of the original case. 

Tunnel2: I added `SOLVER='ULMAT'` to the `PRES` line and removed `TUNNEL_PRECONDITIONER=T`. The ULMAT solver strictly enforces the normal velocity at internal boundaries, in particular your burner. However, the ULMAT solver does not at present run with the tunnel preconditioner option. This case has run out 327 s in about 3 days of run time and is still going. It is running slowly.

Tunnel4: I created a uniform mesh of 0.2 m throughout the entire tunnel using 100 meshes. I also reduced the width of the domain by 0.5 m (-5.4 to 5.6). When using the tunnel preconditioner, it is best to have the least amount of internal blockage as possible. There is no reason to include the lateral solid walls in the simulation. This case is running faster. After 24 h of run time, it has progressed 586 s. My compute cluster is about 8 years old.

There is still the possibility that either case might fail. The HRR is still increasing and the time steps are approximately 0.01 s.
